Wabbajack.App.Wpf/ViewModels/Compiler/CompilerMainVM.cs

```diff
@@ -138,14 +147,66 @@ private async Task Publish()
     {
         _logger.LogInformation("Running preflight checks before publishing list");
         bool readyForPublish = await RunPreflightChecks(CancellationToken.None);
-        if (!readyForPublish) return;
+        if (!readyForPublish)
+            return;
+
+        _logger.LogInformation("Publishing list with machineURL {machineURL}, version {version}, output {output}",
+            Settings.MachineUrl, Settings.Version, Settings.OutputFile);
 
-        _logger.LogInformation("Publishing list with machineURL {machineURL}, version {version}, output {output}", Settings.MachineUrl, Settings.Version, Settings.OutputFile);
         var downloadMetadata = _dtos.Deserialize<DownloadMetadata>(
             await Settings.OutputFile.WithExtension(Ext.Meta).WithExtension(Ext.Json).ReadAllTextAsync())!;
-        await _wjClient.PublishModlist(Settings.MachineUrl, Version.Parse(Settings.Version), Settings.OutputFile, downloadMetadata);
+
+        await Task.Run(async () =>
+        {
+            RxApp.MainThreadScheduler.Schedule(_logger, (_, _) =>
+            {
+                ProgressText = "Publishing...";
+                ProgressState = ProgressState.Normal;
+                ProgressPercent = Percent.Zero;
+                return Disposable.Empty;
+            });
+
+            IsPublishing = true;
+            try
+            {
+                var (progress, publishTask) = await _wjClient.PublishModlist(
+                    Settings.MachineUrl, System.Version.Parse(Settings.Version), Settings.OutputFile, downloadMetadata);
+
+                var progressSubscription = progress
+                    .ObserveOn(RxApp.MainThreadScheduler)
+                    .Subscribe(p =>
+                    {
+                        RxApp.MainThreadScheduler.Schedule(_logger, (_, _) =>
```

Comment:
You're already on the UI thread here by doing ObserveOn(RxApp.MainThreadScheduler). I would replace this with ObserveOnGuiThread to be consistent across the application (it's the same, it's just a WJ-specific shorthand for it)